design/status-design-update.md

```diff
@@ -0,0 +1,1166 @@
+
+```yaml
+status:
+  currentStatus: <Valid|Invalid|Warning>
+  description: <text>
+```
+
+#### `status.includes`
+
+This field should show the status of each include in the `spec.includes` array.
+In order to fully identify them, this means that this must include all the information in each `spec.includes` item, plus the general `status` stanza.
+
+That is, this field should include status of the includes *in this HTTPPRoxy*.
+
+See the examples below to see this in action.
+
+#### `status.routes`
+
+Similarly to `status.includes`, this should include the information from each `route` stanza in `spec.routes`, with an added `status` stanza.
+
+This field should include status of the routes in this HTTPProxy.
+
+#### `status.appliedConditions`
+
+This is where this gets trickier.
+This field is intended to show what conditions are applied when this HTTPProxy object is included from somewhere.
+
+In order to do this, the following information must be present, somehow:
+
+- full list of merged conditions. That is, all prefixes  must be fully merged, and the full set of headers must be visible.
+- a unique identifier for the condition set. In practice, this will end up being related to the inclusion path to this HTTPProxy.
+
+The first is relatively straightforward, a `conditions` array, with the merge rules having been run on it, will suffice.
+
+The second is important for higher-order tool use as well as service admin use.
+
+The exact implementation here will depend on if we do the optional field I include below.
+
+In any case, there should be an `includePath` field in the `status.appliedConditions[n]` object that ensures uniqueness.
+
+This could be rendered as a string, representing the include path, preferably including the FQDN of the root.
+
+So, something that looked like this:
+
+```yaml
+status:
+  appliedConditions:
+    - conditions:
+      - prefix: /foo/bar/baz
+      includePath: "root:www.foobar.com,foo,bar"
+```
+
+Alternatively, if we are not including a more verbose representation of the computed DAG in each object, something like this would be required:
+
+```yaml
+status:
+  appliedConditions:
+    - conditions:
+      - prefix: /foo/bar/baz
+      includePath:
+        name: root
+        namespace: default
+        fqdn: www.foobar.com
+        includes:
+          name: foo
+          namespace: default
+          conditions:
+          - prefix: /foo
+          includes:
+            name: bar
+            namespace: default
+            conditions:
+            - prefix: /bar
+            includes:
+              name: baz
+              namespace: default
+              conditions:
+              - prefix: /baz
+```
+
+#### Optional: `status.proxyGraph`
+
+The intent of this field is to store enough breadcrumbs to allow a user (or external tool) to be able to deduce the entire HTTPProxy graph (if they have RBAC access to see the objects in all referenced namespaces).
```

Comment:
Can't we just embed that in status directly via the name/namespace of objects in the include path?